src/components/Movie.jsx

```diff
@@ -0,0 +1,72 @@
+import { useDispatch, useSelector } from 'react-redux'
+import starredSlice from '../data/starredSlice'
+import watchLaterSlice from '../data/watchLaterSlice'
+import placeholder from '../assets/not-found-500X750.jpeg'
+
+const Movie = ({ movie, viewTrailer, closeCard }) => {
+
+    const state = useSelector((state) => state)
+    const { starred, watchLater } = state
+    const { starMovie, unstarMovie } = starredSlice.actions
+    const { addToWatchLater, removeFromWatchLater } = watchLaterSlice.actions
+
+    const dispatch = useDispatch()
+
+    const myClickHandler = (e) => {
+        if (!e) var e = window.event
+        e.cancelBubble = true
+        if (e.stopPropagation) e.stopPropagation()
+        e.target.parentElement.parentElement.classList.remove('opened')
+    }
+
+    return (
+        <div className="wrapper col-3 col-sm-4 col-md-3 col-lg-3 col-xl-2">
+        <div className="card" onClick={(e) => e.currentTarget.classList.add('opened')} >
+            <div className="card-body text-center">
+                <div className="overlay" />
+                <div className="info_panel">
+                    <div className="overview">{movie.overview}</div>
+                    <div className="year">{movie.release_date?.substring(0, 4)}</div>
+                    {!starred.starredMovies.map(movie => movie.id).includes(movie.id) ? (
```

Comment:
there's an opportunity to enhance the code by minimizing the unnecessary creation of an array of IDs.

```
const isStarred = starred.starredMovies.some(starredMovie => starredMovie.id === movie.id);
```